tee-worker/service/src/sync_state.rs

```diff
@@ -42,13 +44,10 @@ pub(crate) fn sync_state<
 	enclave_api: &E,
 	skip_ra: bool,
 ) {
-	// FIXME: we now assume that keys are equal for all shards.
 	let provider_url = match WorkerModeProvider::worker_mode() {
-		WorkerMode::Sidechain =>
+		WorkerMode::Sidechain | WorkerMode::OffChainWorker =>
 			executor::block_on(get_author_url_of_last_finalized_sidechain_block(node_api, shard))
```

Comment:
How is it possible to get author of last finalized sidechain block in offchain mode ?

Reply:
Good catch - it's the result of patch apply from upstream. By taking a closer look at the code, it seems that it's not properly named: internally it's calling `primary_worker_for_shard`

I can submit an upstream issue for it.